pkg/devconsole/tekton-results/results.go

```diff
@@ -0,0 +1,299 @@
+	if err != nil {
+		return "", err
+	}
+	if localHost {
+		route, err := dynamicClient.Resource(*TektonResultsAPIRoute).Namespace("openshift-pipelines").Get(context.TODO(), "tekton-results-api-service", metav1.GetOptions{})
+		if err != nil {
+			return "", err
+		}
+		cachedTektonResultsHost, isHostPresent, err := unstructured.NestedString(route.Object, "spec", "host")
+		if err != nil || !isHostPresent {
+			return "", err
+		}
+		return cachedTektonResultsHost, nil
+	}
+
+	if cachedTektonResultsHost != "" {
+		return cachedTektonResultsHost, nil
+	}
+
+	host, err := dynamicClient.Resource(*TektonResultsResource).Get(context.TODO(), "result", metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+	targetNamespace, isTargetNsPresent, err := unstructured.NestedString(host.Object, "spec", "targetNamespace")
+	if err != nil || !isTargetNsPresent {
+		targetNamespace = ""
+	}
+	serverPort, isPortPresent, err := unstructured.NestedString(host.Object, "spec", "server_port")
+	if err != nil || !isPortPresent {
+		serverPort = "8080"
+	}
+	tlsHostname, isTLSHostnamePresent, err := unstructured.NestedString(host.Object, "spec", "tls_hostname_override")
+	if err != nil || !isTLSHostnamePresent {
+		tlsHostname = ""
+	}
+
+	if tlsHostname != "" {
+		cachedTektonResultsHost = fmt.Sprintf("%s:%s", tlsHostname, serverPort)
+	} else if targetNamespace != "" && serverPort != "" {
+		cachedTektonResultsHost = fmt.Sprintf("tekton-results-api-service.%s.svc.cluster.local:%s", targetNamespace, serverPort)
+	} else {
+		cachedTektonResultsHost = fmt.Sprintf("tekton-results-api-service.openshift-pipelines.svc.cluster.local:%s", serverPort)
+	}
+	return cachedTektonResultsHost, nil
+}
+
+func GetTektonResults(r *http.Request, user *auth.User, dynamicClient *dynamic.DynamicClient) (common.DevConsoleCommonResponse, error) {
+	var request TektonResultsRequest
+	err := json.NewDecoder(r.Body).Decode(&request)
+	if err != nil {
+		return common.DevConsoleCommonResponse{}, fmt.Errorf("failed to parse request: %v", err)
+	}
+
+	TEKTON_RESULTS_HOST, err := getTRHost(r, dynamicClient)
+	if err != nil {
+		return common.DevConsoleCommonResponse{}, fmt.Errorf("failed to get TektonResults host: %v", err)
+	}
+
+	TEKTON_RESULTS_URL := fmt.Sprintf("https://%s/apis/results.tekton.dev/v1alpha2/parents/%s/results/-/records?%s", TEKTON_RESULTS_HOST, request.SearchNamespace, request.SearchParams)
+
+	var serviceRequest *http.Request
+	serviceRequest, err = http.NewRequest(http.MethodGet, TEKTON_RESULTS_URL, nil)
+
+	if err != nil {
+		return common.DevConsoleCommonResponse{}, fmt.Errorf("failed to create request: %v", err)
+	}
+
+	if request.AllowAuthHeader {
+		serviceRequest.Header.Set("Authorization", fmt.Sprintf("Bearer %s", user.Token))
+	}
+
+	var serviceTransport *http.Transport
+	if request.AllowInsecure {
+		serviceTransport = &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		}
```

Comment:
Same question on allowing insecure requests. What is the use case? Is this only for development?

Also for the other endpoints below.

Reply:
For tekton results, we found that we require this mandatorily.

Reply:
Why do we require this? Can we not use service serving certificates or another CA cert?

We should never use `InsecureSkipVerify` in a production environment.

Reply:
Finally, removed it. Used Certs :)